runtime/binding-http/src/main/java/io/aklivity/zilla/runtime/binding/http/internal/stream/HttpClientFactory.java

```diff
@@ -2094,7 +2103,7 @@ private HttpClient supplyClient()
                 client = clients.stream().filter(
                         c -> !HttpState.replyOpened(c.state) ||
                         c.encoder == HttpEncoder.HTTP_2 ||
-                        c.exchange == null && c.encoder == HttpEncoder.HTTP_1_1)
+                        c.exchange == null && c.encoder == HttpEncoder.HTTP_1_1 || c.encoder == HttpEncoder.H2C)
```

Comment:
Let's put each `||` logical or on a separate line for clarity, since the `&&` logical and is more tightly binding.